### PR TITLE
keysend: enforce BOLT11 description length limit

### PR DESCRIPTION
**Pull request by noblepayne**
The keysend plugin uses `> 1023` as the cutoff for description length when inserting an invoice. This was inconsistent with `lightnind/invoice.c`, which enforces the BOLT11 description field limit defined in `common/bolt11.h`.

This patch switches to using `BOLT11_FIELD_BYTE_LIMIT` directly. As a result, keysend no longer fails on descriptions between 641–1023 bytes, which previously caused unexpected failures.

A new regression test (`test_keysend_description_size_limit`) exercises boundary cases just below, at, and above the limit.

Changelog-None

## Checklist
Before submitting the PR, ensure the following tasks are completed. If an item is not applicable to your PR, please mark it as checked:

- [x] The changelog has been updated in the relevant commit(s) according to the [guidelines](https://docs.corelightning.org/docs/coding-style-guidelines#changelog-entries-in-commit-messages).
- [x] Tests have been added or modified to reflect the changes.
- [x] Documentation has been reviewed and updated as needed.
- [x] Related issues have been listed and linked, including any that this PR closes.

### CODE REVIEW

File: plugins/keysend.c [hunk 1]
```diff
@@ -563,7 +564,7 @@ static struct command_result *htlc_accepted_call(struct command *cmd,
 					   (const char *)desc_field->value);
 		json_add_string(req->js, "description", desc);
 		/* Don't exceed max possible desc length! */
-		if (strlen(desc) > 1023)
+		if (strlen(desc) >= BOLT11_FIELD_BYTE_LIMIT)
```
Comment (noblepayne) [inline plugins/keysend.c]: Should this be `>` or `>=`? [`lightningd/invoice.c`](https://github.com/ElementsProject/lightning/blob/master/lightningd/invoice.c#L1138) uses `>`, however the [BOLT11 spec](https://github.com/lightning/bolts/blob/master/11-payment-encoding.md#tagged-fields) says:
> Note that the maximum length of a Tagged Field's data is constricted by the maximum value of data_length. This is 1023 x 5 bits, or 639 bytes.

In [`common/bolt11.c`](https://github.com/ElementsProject/lightning/blob/master/common/bolt11.h#L12) `BOLT11_FIELD_BYTE_LIMIT` has a value of 640.

Far from an expert here and appreciate any insight. Happy to update as needed.

Comment (rustyrussell) [reply plugins/keysend.c]: Hmm, we should fix invoice.c too.  Let me write a test.

**Comment by rustyrussell** [comment]
OK, I fixed the invoice 640-byte desc error you found!  And I modified the Changelog-None to a proper changelog line for your fix.

Thanks!
